backend/src/lib/passport.js

```diff
@@ -0,0 +1,51 @@
+const passport = require('passport');
+const LocalStrategy = require('passport-local');
+const JwtStrategy = require('passport-jwt').Strategy;
+const ExtractJwt = require('passport-jwt').ExtractJwt;
+const CONFIG = require('../../../config/config.json');
+const klad = require('../../../domain/src/main');
+const User = require('../../../domain/src/User');
+
+
+passport.use(new LocalStrategy(
+    {
+        usernameField: 'email',
+        passwordField: 'password',
+        session: false
+    },
+    async function (email, password, done) {
+        let user,
+            err = null;
+        try {
+            user = await klad.getByPropFromKladovka('users', 'email', email);
+            //typeof user._id == object
+            if (!user || !User.comparePasswords(user, password))
+                user = false;
+        } catch (err) {
+            return done(err);
+        }
+        return done(err, user);
+    })
+);
+
+passport.use(new JwtStrategy(
+    {
+        jwtFromRequest: ExtractJwt.fromAuthHeaderWithScheme('Bearer'),
+        secretOrKey: CONFIG.JWT_SECRET
+    },
+    async function (payload, done) {
+        let user,
+            err = null;
+        try {
+            user = await klad.getFromKladovka('users', payload._id);
+            if (user !== null)
+                user._id = user._id.toString();
```

Comment:
wtf, tried to reply with email twice, no luck :man_shrugging: 
anyway, mongo does some strange things sometimes, but this is not the right place to do that conversion. First of all, why do you even need that conversion? Why don't you just leave it the way mongo wants it? Second, even if you need so badly that conversion, it should happen on the db layer, not here.